models/classes/runner/QtiRunnerService.php

```diff
@@ -1195,6 +1204,12 @@ public function finish(RunnerServiceContext $context, $finalState = DeliveryExec
                 $result = false;
             }
 
+            /** @var ToolsStateStorage $toolsStateStorage */
+            $toolsStateStorage = $this->getServiceManager()->get(ToolsStateStorage::SERVICE_ID);
+            $toolsStates = $toolsStateStorage->deleteStates($context->getTestExecutionUri());
```

Comment:
Before doing any query you should verify if the config contains tools states to be synchronized